narwhal/primary/src/state_handler.rs

```diff
@@ -62,30 +63,46 @@ pub struct StateHandler {
 }
 
 // Internal state for randomness DKG and generation.
-// TODO: Write a brief protocol description.
+//
+// DKG protocol:
+// 1. This node sends out a `Message` to all other nodes.
+// 2. Once sufficient valid `Message`s are received from other nodes via consensus and processed,
+//    this node sends out a `Confirmation` to all other nodes.
```

Comment:
Which step writes the used message?

Reply:
IIUC this bookkeeping is done just before sending out the confirmation.

Reply:
Right, from perspective of code in state handler it's just a black box we record bc fastcrypto needs us to pass it back in later